games-emulation/stella/stella-5.1.3.ebuild

```diff
@@ -0,0 +1,59 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit desktop gnome2-utils
+
+DESCRIPTION="Multi-platform Atari 2600 VCS Emulator"
+HOMEPAGE="https://stella-emu.github.io"
+SRC_URI="https://github.com/stella-emu/${PN}/releases/download/${PV}/${P}-src.tar.xz"
+
+LICENSE="GPL-2+ BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="joystick"
+
+RDEPEND="
+	media-libs/libsdl2[joystick?,opengl,video]
+	media-libs/libpng:0=
+	sys-libs/zlib
```

Comment:
you do not need zlib explicitly

Reply:
Removed from the ebuild.

I can't deal with bundling though, beyond my expertise.

At any rate the bundling issue is probably an upstream call.

Reply:
You mean version 4.7.2?

Reply:
yes

Reply:
Yes it was bundled in 4.7.2, I just checked the tarball.  Why do you ask?

Reply:
Actually I just asked upstream to unbundle them and they told me something very interesting:

https://github.com/stella-emu/stella/issues/391

The bundled versions are only used as fallbacks if the system versions are not installed.  If they exist on the system they will be used.

Reply:
So this means that we *should* depend on them in the ebuild because at build time they will be used over the bundled versions if they're installed.

Reply:
That should be obvious by looking at the build system then, did you not?

Reply:
Admittedly not.

I was under the impression that the best thing I could do was copy the previous ebuild to the new version number and screw with it as little as possible.  I'm not competent enough yet for a whole-scale overhaul of the ebuilds.

So it never occurred to me to dig into it.

As it is given how my bug reports in general have been handled I was under the impression that digging into things like that was upstream's job.

Reply:
When you change dependencies, you look at the build system to check if that is valid.
When you bump versions, you look at the change log and build system for changes.

Reply:
Can I trust the people who wrote the previous ebuild or does every new version need a complete verification from scratch?

As for packaging, upstream informed me that they bundled zlib and libpng intentionally.  You do have a point though that I should have noticed.

Reply:
You can't trust the people who wrote the previous ebuild to be aware of changes upstream made in future releases, is my point.

The bundling is fine as long as we can make sure it is not in effect with our package. To make sure of that, ebuilds are often removing those directories where the bundled libs reside.